client/signup/steps/new-user-survey/components/Shuffle/test/index.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render } from '@testing-library/react';
+import React, { ReactNode, useState } from 'react';
+import Shuffle from '../index';
+interface MockComponentProps {
+	children?: ReactNode[];
+	childOrderOverride?: string[] | null;
+	ref?: React.Ref< any >;
+	isShuffleActive?: boolean;
+}
+const DEBUG_PRINT_DIV_ID = 'render-order-print';
+function MockComponent( { children, childOrderOverride, isShuffleActive }: MockComponentProps ) {
+	const [ childOrder, setChildOrder ] = useState< string[] | null >( childOrderOverride ?? null );
+	if ( children ) {
+		return (
+			<>
+				<div data-testid={ DEBUG_PRINT_DIV_ID }>{ childOrder?.join( ',' ) }</div>
+				<Shuffle
+					isShuffleActive={ isShuffleActive }
+					childOrder={ childOrder }
+					getChildKey={ ( child: React.ReactNode ) => {
+						// @ts-expect-error - Resolver does not have to have proper form
+						return child.props.children[ 1 ].props.name ?? '';
+					} }
+					setChildOrder={ setChildOrder }
+				>
+					{ children }
+				</Shuffle>
+			</>
+		);
+	}
+	return (
+		<>
+			<div data-testid={ DEBUG_PRINT_DIV_ID }>{ childOrder?.join( ',' ) }</div>
+			<Shuffle
+				childOrder={ childOrder }
+				setChildOrder={ setChildOrder }
+				getChildKey={ ( child: any ) => {
+					return child.props.children[ 1 ].props.name;
+				} }
+				isShuffleActive={ isShuffleActive }
+			>
+				<label key={ 1 }>
+					<span>Child 1</span>
+					<input name="1" />
+				</label>
+				<label key={ 2 }>
+					<span>Child 2</span>
+					<input name="2" />
+				</label>
+				<label key={ 3 }>
+					<span>Child 3</span>
+					<input name="3" />
+				</label>
+				<label key={ 4 }>
+					<span>Child 4</span>
+					<input name="4" />
+				</label>
+				<label key={ 5 }>
+					<span>Child 5</span>
+					<input name="5" />
+				</label>
+			</Shuffle>
+		</>
+	);
+}
+
+describe( 'Shuffle Component', () => {
+	test( 'Given an order of ids renders according to the given order', () => {
+		const { getByTestId } = render(
+			<MockComponent childOrderOverride={ [ '5', '4', '3', '2', '1' ] } isShuffleActive={ true } />
+		);
+		expect( getByTestId( '1' ) ).toHaveTextContent( 'Child 5' );
+		expect( getByTestId( '2' ) ).toHaveTextContent( 'Child 4' );
+		expect( getByTestId( '3' ) ).toHaveTextContent( 'Child 3' );
+		expect( getByTestId( '4' ) ).toHaveTextContent( 'Child 2' );
+		expect( getByTestId( '5' ) ).toHaveTextContent( 'Child 1' );
```

Comment:
Another approach would be to avoid using test-id's and instead:

```
...

const labels = screen.getAllByText( /child/i );

expect( labels[ 0 ] ).toHaveTextContent( 'Child 5' );
expect( labels[ 1 ] ).toHaveTextContent( 'Child 4' );
expect( labels[ 2 ] ).toHaveTextContent( 'Child 3' );
expect( labels[ 3 ] ).toHaveTextContent( 'Child 2' );
expect( labels[ 4 ] ).toHaveTextContent( 'Child 1' );
```